src/main/java/net/openhft/chronicle/queue/RollCycles.java

```diff
@@ -15,12 +15,14 @@
  */
 package net.openhft.chronicle.queue;
 
+import org.jetbrains.annotations.NotNull;
+
 import net.openhft.chronicle.core.Maths;
 import net.openhft.chronicle.core.time.TimeProvider;
-import org.jetbrains.annotations.NotNull;
 
 public enum RollCycles implements RollCycle {
     TEST_SECONDLY(/*--*/"yyyyMMdd-HHmmss", 1000, 1 << 15, 4), // only good for testing
+    MILLI_WEEKLY(/*---*/"yyyyMMdd-HHmmssSSS", 7 * 24 * 60 * 60 * 1000, 4 << 10, 128), // 4 trillion entries per milli-week
```

Comment:
Isn't a MILLI_WEEK about 10 minutes? If so, should go beneath minutes. I don't understand why pattern is not same as MINUTELY or am I not getting it?